app/grandchallenge/cases/tasks.py

```diff
@@ -208,27 +208,25 @@ def build_images(  # noqa:C901
             error_message="Time limit exceeded",
         )
     except Exception:
-        _delete_session_files(upload_session=upload_session)
         error_handler.handle_error(
             interface=ci,
             error_message="An unexpected error occurred",
         )
         logger.error("An unexpected error occurred", exc_info=True)
-
-
-def _handle_raw_image_files(*, tmp_dir, upload_session):
-    importer_result = import_images(
-        input_directory=tmp_dir, origin=upload_session
-    )
-
-    _handle_raw_files(
-        consumed_files=importer_result.consumed_files,
-        file_errors=importer_result.file_errors,
-        base_directory=tmp_dir,
-        upload_session=upload_session,
-    )
-
-    _delete_session_files(upload_session=upload_session)
+    finally:
+        upload_session.user_uploads.all().delete()
+
+    if (
+        len(upload_session.import_result["file_errors"]) != 0
+        and len(upload_session.import_result["consumed_files"]) == 0
```

Comment:
Why do we need the check on both? I would have thought that `len(upload_session.import_result["consumed_files"]) == 0` would be enough?

Reply:
True, I wasn't sure, and went the save route in checking both, but I think you're right that there will always be file errors if there are 0 consumed files, so just checking for the consumed files should be enough.